tardis/simulation/base.py

```diff
@@ -441,11 +340,11 @@ def run(self):
                 self.model.t_inner,
             )
             self.iterate(self.no_of_packets)
-            self.converged = self.advance_state()
+            self.advance_state()
             if hasattr(self, "convergence_plots"):
                 self.convergence_plots.update()
             self._call_back()
-            if self.converged:
+            if self.convergence_strategy.converged:
                 if self.convergence_strategy.stop_if_converged:
```

Comment:
Perhaps this should be a single conditional and the convergence_strategy should have a separate single attribute that tells it to stop or not based on whether or not stop_if_converged is true.

Reply:
I see. I'll add a new @property "should_stop = self.converged and self.stop_if_converged" in the ConvergenceStrategy class. We'll only check this property in the simulation class above.

If you can think of a better name than "should_stop", please let me know

Reply:
@Rodot- I've handled this, please review

I also wanted to ask, what test cases can I add for this PR?